server/server.go

```diff
@@ -861,11 +863,11 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	registerDownloadHandlers(mux, "/download")
 
 	// Serve extensions
-	var extensionsApiPath = "/extensions/"
 	var extensionsSharedPath = "/tmp/extensions/"
 
-	extHandler := http.StripPrefix(extensionsApiPath, http.FileServer(http.Dir(extensionsSharedPath)))
-	mux.HandleFunc(extensionsApiPath, extHandler.ServeHTTP)
+	mux.HandleFunc("/extensions.js", func(writer http.ResponseWriter, _ *http.Request) {
```

Comment:
Should this endpoint require auth? The extension might contain sensitive system design information. (Or, in the worse case, secrets - people do not-smart things. :-P)

Reply:
I think extensions are just code so should be handled the same as any other javascript file: User should be able to load it without authentication and no secrets should be hardcoded in extension.